src/main/DAL/Exam/ExamRepository.js

```diff
@@ -0,0 +1,131 @@
+        this.#Question.belongsTo(this.#MetaQuestion, { as: 'metaQuestion', foreignKey: 'metaQuestionId' });
+
+        // Question - Answer, many to many
+        this.#Question.belongsToMany(this.#Answer, { as: 'answers', through: this.#QuestionAnswer });
+        this.#Answer.belongsToMany(this.#Question, { as: 'questions', through: this.#QuestionAnswer });
+    }
+    /**
+     * 
+     * @param  examData { title:string }
+     * @returns New Dal exam promise
+     */
+    async createExam(examData) {
+        return await this.#Exam.create(examData);
+    }
+
+    /**
+     * @returns exam object
+     */
+    async getExamById(id) {
+        try {
+            return await this.#Exam.findByPk(id, {
+                include: [
+                    {
+                        model: this.#Question,
+                        as: 'questions',
+                        include: [
+                            {
+                                model: this.#MetaQuestion,
+                                as: 'metaQuestion',
+                            },
+                            {
+                                model: this.#Answer,
+                                as: 'answers',
+                            }
+                        ]
+                    },
+                ]
+            });
+        }catch (e) {
+            console.error(e);
+            throw EMSError(EXAM_PROCESS_ERROR_MSGS.GET_EXAM_BY_ID(id), EXAM_PROCESS_ERROR_CODES.GET_EXAM_BY_ID)
+        }
+    }
+    /**
+     * 
+     * @param  examId 
+     * @param  mQid : the metaQuestion id of the question
+     * @param  questionData : {ordinal : the ordinal of the question in the exam}
+     * @param  answerOrdinalObj : an array of object, maps the answerId to its ordinal in the question. type [{id, ordinal}]
+     * @returns The new question
+     */
+    async addQuestionToExam(examId, mQid, questionData, answerIdToOrdinalArr) {
+        try {
+            const exam = await this.getExamById(examId);
+            const question = await this.#Question.create(questionData);
+            const metaQuestion = await this.#MetaQuestion.findByPk(mQid);
+            await question.setMetaQuestion(metaQuestion);
+            await question.setExam(exam);
+            await this.#setAnswersToQuestion(question.id, answerIdToOrdinalArr);
+            return await this.#Question.findByPk(question.id, {
+                include: [
+                    {
+                        model: this.#MetaQuestion,
+                        as: 'metaQuestion',
+                    },
+                    {
+                        model: this.#Answer,
+                        as: 'answers',
+                    },
+                ]
+            });
+        } catch (e) {
+            console.error(e);
+            throw EMSError(EXAM_PROCESS_ERROR_MSGS.ADD_QUESTION_TO_EXAM(examId, mQid, questionData, answerIdToOrdinalArr), EXAM_PROCESS_ERROR_CODES.ADD_QUESTION_TO_EXAM)
+        }
```

Comment:
this try/catch is kinda bad, I want better detail and not guess what went wrong. 
if this failed because the `mqId` was invalid then the error code should match and give that information
if the `examId` was invalid... 
and so on